pandas/core/tools/datetimes.py

```diff
@@ -28,13 +29,58 @@ def _guess_datetime_format_for_array(arr, **kwargs):
         return _guess_datetime_format(arr[non_nan_elements[0]], **kwargs)
 
 
+def should_cache(arg, unique_share=0.7, check_count=None):
+    """
+    Decides whether to do caching.
+
+    If the percent of unique elements among `check_count` elements less
+    than `unique_share * 100` then we can do caching.
+
+    Parameters
+    ----------
+    arg: listlike, tuple, 1-d array, Series
+    unique_share: float or None
+        0 < unique_share < 1
+    check_count: int or None
+        0 <= check_count <= len(arg)
+
+    Returns
+    -------
+    do_caching: bool
+    """
+    do_caching = True
+
+    # default realization
```

Comment:
can you give a 1 or 2 line explanation of the strategy of determining whether we should cache here (to the extent it differs from the doc-string)